tests/test_ops.py

```diff
@@ -131,60 +131,29 @@ def test_to_ops_stencil(self, _accesses):
             Literal('"%s"' % stencil_name.upper())
         )
 
-    def test_create_ops_dat_time_function(self):
-        grid = Grid(shape=(4))
-
-        u = TimeFunction(name='u', grid=grid, space_order=2)
-
-        block = OpsBlock('block')
-
-        name_to_ops_dat = {}
-
-        result = create_ops_dat(u, name_to_ops_dat, block)
-
-        assert name_to_ops_dat['ut0'].base.name == namespace['ops_dat_name'](u.name)
-        assert name_to_ops_dat['ut0'].indices == (Symbol('t0'),)
-        assert name_to_ops_dat['ut1'].base.name == namespace['ops_dat_name'](u.name)
-        assert name_to_ops_dat['ut1'].indices == (Symbol('t1'),)
-
-        assert result[0].expr.lhs.name == namespace['ops_dat_dim'](u.name)
-        assert result[0].expr.rhs.params == (Integer(4),)
+    @pytest.mark.parametrize('equation,expected', [
+        ('Eq(u.forward, u + 1)',
+         '[\'ops_dat u_dat[2] = {ops_decl_dat(block, 1, u_dim, u_base, u_d_m, u_d_p, '
+         'u[0], "float", "ut0"), ops_decl_dat(block, 1, u_dim, u_base, u_d_m, u_d_p, '
+         'u[1], "float", "ut1")}\']'),
+        ('Eq(u.forward, u + v.dx)',
+         '[\'ops_dat u_dat[2] = {ops_decl_dat(block, 1, u_dim, u_base, u_d_m, u_d_p, '
+         'u[0], "float", "ut0"), ops_decl_dat(block, 1, u_dim, u_base, u_d_m, u_d_p, '
+         'u[1], "float", "ut1")}\','
+         '\'ops_dat v_dat;\','
+         '\'v_dat = ops_decl_dat(block, 1, v_dim, v_base, v_d_m, v_d_p, '
+         'v[0], "float", "v")\']')
+    ])
+    def test_create_ops_dat(self, equation, expected):
```

Comment:
this test seems much thinner than before?

Reply:
Although it is thinner, it is actually testing more things than before.
The first test in the `@pytest.mark.parametrize` cover the case that was being tested, and I added the second one to test a mix of `TimeFunction` and `Function`.

Reply:
oki